docs/general/developer-resources/emerald-paratime/integrating-band-oracle-smart-contract.md

```diff
@@ -19,12 +19,40 @@ contracts. You can read more about the specific details of the protocol
 2. Compile the contract with compiler version `0.6.11`.
 3. Switch to the Deploy tab of Remix.
    1. Select "Injected Web3" in the Environment dropdown in the top left to connect Metamask.
-   2. Make sure that Metamask is connected to the Alfajores test network. You can read about adding Alfajores to Metamask [here](https://docs.celo.org/getting-started/wallets/using-metamask-with-celo/manual-setup#adding-a-celo-network-to-metamask).
+   2. Make sure that Metamask is connected to the Emerald (Testnet/Mainnet) network. You can read about adding Emerald network to Metamask [here](https://docs.oasis.dev/general/manage-tokens/how-to-transfer-rose-into-emerald-paratime#metamask).
 
-![environment](../../images/emerald/band-remix-environment.png)
+![environment](../../images/emerald/band_demooracle_smartcontract.png)
 
-1. Enter the Alfajores testnet Band reference data aggregator contract address (`0x71046b955Cdd96bC54aCa5E66fd69cfb5780f3BB`) to the `DemoOracle` constructor and deploy the contract. You can access the reference data aggregator contract on mainnet at `0xDA7a001b254CD22e46d3eAB04d937489c93174C3`.
+4. Enter the Emerald Testnet Band reference data aggregator contract address (`0x61704EFB8b8120c03C210cAC5f5193BF8c80852a`) to the `DemoOracle` constructor and deploy the contract. You can access the reference data aggregator contract on mainnet at `0xDA7a001b254CD22e46d3eAB04d937489c93174C3`.
 
-![environment](../../images/emerald/band-remix-deploy.png)
+![environment](../../images/emerald/band_deploy_demooracle_smartcontact.png)
 
 An interface to interact with the contract will appear in the bottom left corner of Remix.
+
+### Get Rates {#get-rates}
+
+Clicking the `getPrice` button will return the current price of WBTC in USD. This function calls `getReferenceData(string memory _base, string memory _quote)` on the Band reference data contract, passing "WBTC" and "USD", indicating WBTC as the base and USD as the quote. The rate returned is base/quote multiplied by 1e18.
+
+![environment](../../images/emerald/band_get_rates.png)
+
+Note that the `DemoOracle` contract only returns the latest rate, but the reference contract also returns values of the last time the base and quote references were updated.
+
+The price is offset by 1e18. The returned value at the time of testing is `395670000000000000`. Multiplying by 1e-18 gives the current USD price given by the reference contract, 39567.00 WBTC/USD.
```

Comment:
```suggestion
The price is offset by 1e18. The returned value at the time of testing is `395670000000000000`. Multiplying by 1e-18 gives the current USD price given by the reference contract, 39567.00 USD/WBTC.
```

Reply:
Also, 395670000000000000 * 1e-18 = 0.39567. How did you get 39,567.00?

Reply:
Updated number 395670000000000000 to 39567000000000000000000 * 1e-18 = 39,567.00.